vault/src/metrics.rs

```diff
@@ -376,9 +426,20 @@ fn publish_bitcoin_balance<B: BitcoinCoreApi + Clone + Send + Sync>(vault: &Vaul
     }
 }
 
-async fn publish_native_currency_balance(parachain_rpc: &InterBtcParachain) {
-    if let Ok(balance) = parachain_rpc.get_free_balance(parachain_rpc.native_currency_id).await {
-        let balance = raw_value_as_currency(balance, parachain_rpc.native_currency_id);
+async fn publish_native_currency_balance<
+    P: VaultRegistryPallet
+        + CollateralBalancesPallet
+        + RedeemPallet
+        + IssuePallet
+        + ReplacePallet
+        + RefundPallet
+        + UtilFuncs,
+>(
+    parachain_rpc: &P,
+    native_currency: CurrencyId,
```

Comment:
Instead of passing this is an argument, maybe we should add `get_native_currency()` to the `UtilFuncs` trait